### PR TITLE
Introduce sections in usage help

### PR DESCRIPTION
**Pull request by wladekb**
I've seen a bunch of unordered arguments in the usage help. It is tough to find anything there. Hope this change helps with its readability and usability.

### CODE REVIEW

**Comment by macbre** [comment]
LGTM, thanks @wladekb !

File: lib/optimist-config-file.js [hunk 1]
```diff
@@ -106,7 +106,42 @@ optimist.parse = function(args) {
...
+optimist.help = function() {
+	var text = _help.apply(optimist, arguments),
+		i, s, p, opt_header = "\nOptions:",
+		opt_header_len = opt_header.length;
+	//console.log(text,headers);
+	for (i = 0; i < headers.length; i++) {
+		s = "\n  --" + headers[i][1];
+		p = text.indexOf(s);
+		if (p != -1) {
+			s = "\n" + headers[i][0] + ":";
```
Comment (macbre) [inline lib/optimist-config-file.js]: What if we emit an empty line before the options group name and make it "bold" by calling `bright` from `ansistyles` module?

Comment (wladekb) [inline lib/optimist-config-file.js]: I wasn't aware of any `bright` option. How this could be fixed?

Comment (macbre) [inline lib/optimist-config-file.js]:
```js
s = "\n" + ansistyles.bright(headers[i][0]) + ":";
```

and of course require `ansistyles` :)

**Comment by macbre** [comment]
Thank you, @wladekb

**Pull request merged**